pygmt/tests/test_solar.py

```diff
@@ -0,0 +1,120 @@
+"""
+Tests for solar.
+"""
+import datetime
+
+import pytest
+from pygmt import Figure
+from pygmt.exceptions import GMTInvalidInput
+
+
+@pytest.mark.mpl_image_compare
+def test_solar_terminators():
+    """
+    Test passing the solar argument with a time string and no terminator type
+    to confirm the default terminator type.
+    """
+    fig = Figure()
+    fig.basemap(region="d", projection="W0/15c", frame="a")
+    fig.solar(
+        terminator="d",
+        pen="1p,blue",
+        terminator_datetime="1990-02-17 04:25:00",
+    )
+    fig.solar(
+        terminator="a",
+        pen="1p,red",
+        terminator_datetime="1990-02-17 04:25:00",
+    )
+    fig.solar(
+        terminator="c",
+        pen="1p,green",
+        terminator_datetime="1990-02-17 04:25:00",
+    )
+    fig.solar(
+        terminator="n",
+        pen="1p,yellow",
+        terminator_datetime="1990-02-17 04:25:00",
+    )
+    return fig
+
+
+@pytest.mark.mpl_image_compare(filename="test_solar_set_terminator_datetime.png")
+@pytest.mark.parametrize(
+    "terminator_datetime",
+    [
+        "1990-02-17T04:25:00",
```

Comment:
Ok, we can set a pytest.param id, so that there won't be a space in the filename, nor will there be colons that Windows doesn't like.
```suggestion
        pytest.param("1990-02-17 04:25:00", id="terminator_datetime0"),
```